apps/client/src/liquidityVenues/midas/index.ts

```diff
@@ -0,0 +1,329 @@
+      ONE_HUNDRED_PERCENT,
+    );
+
+    return (params.amountMTokenIn * feePercent) / ONE_HUNDRED_PERCENT;
+  }
+
+  private _requireAndUpdateLimit(params: PreviewRedeemInstantParams, amount: bigint) {
+    return params.dailyLimits + amount <= params.instantDailyLimit;
+  }
+
+  private _convertMTokenToUsd(params: PreviewRedeemInstantParams, amount: bigint) {
+    if (amount === 0n || params.mTokenRate === 0n) return undefined;
+
+    return {
+      amountUsd: (amount * params.mTokenRate) / 10n ** 18n,
+      mTokenRate: params.mTokenRate,
+    };
+  }
+
+  private _convertUsdToToken(params: PreviewRedeemInstantParams, amountUsd: bigint) {
+    if (amountUsd === 0n) return undefined;
+
+    const tokenRate = params.tokenOutConfig.stable ? params.STABLECOIN_RATE : params.tokenOutRate;
+
+    if (tokenRate === 0n) return undefined;
+
+    return {
+      amountToken: (amountUsd * 10n ** 18n) / tokenRate,
+      tokenRate,
+    };
+  }
+
+  private _truncate(value: bigint, decimals: bigint) {
+    return this._convertToBase18(this._convertFromBase18(value, decimals), decimals);
+  }
+
+  private _convertFromBase18(originalAmount: bigint, decidedDecimals: bigint) {
+    return this._convert(originalAmount, 18n, decidedDecimals);
+  }
+
+  private _convertToBase18(originalAmount: bigint, originalDecimals: bigint) {
+    return this._convert(originalAmount, originalDecimals, 18n);
+  }
+
+  private _convert(originalAmount: bigint, originalDecimals: bigint, decidedDecimals: bigint) {
+    if (originalAmount === 0n) return 0n;
+    if (originalDecimals === decidedDecimals) return originalAmount;
+
+    let adjustedAmount = 0n;
+
+    if (originalDecimals > decidedDecimals) {
+      adjustedAmount = originalAmount / 10n ** (originalDecimals - decidedDecimals);
+    } else {
+      adjustedAmount = originalAmount * 10n ** (decidedDecimals - originalDecimals);
+    }
+
+    return adjustedAmount;
+  }
+
+  // async methods
+
+  async getRedemptionParams(
+    vault: Address,
+    tokenOut: Address,
+    seizedCollateral: bigint,
+    encoder: ExecutorEncoder,
+  ): Promise<PreviewRedeemInstantParams | undefined> {
+    try {
+      const [
+        minAmount,
+        instantFee,
+        instantDailyLimit,
+        STABLECOIN_RATE,
+        waivedFeeRestriction,
+        dailyLimits,
+        mTokenDataFeed,
+        tokenOutConfig,
+        tokenOutDecimals,
+      ] = await Promise.all([
+        this.getRedemptionVaultMinAmount(vault, encoder),
+        this.getRedemptionVaultInstantFee(vault, encoder),
+        this.getRedemptionVaultInstantDailyLimit(vault, encoder),
+        this.getRedemptionVaultStableCoinRate(vault, encoder),
+        this.getRedemptionVaultWaivedFeeRestriction(vault, encoder),
+        this.getRedemptionVaultDailyLimits(vault, encoder),
+        this.getRedemptionVaultmTokenDataFeed(vault, encoder),
+        this.getRedemptionVaultMTokenOutConfig(vault, tokenOut, encoder),
```

Comment:
Let's try to use viem functionality when possible, as it can be hard to read with so many helper functions. Recommendation:

Above this, do `const midasContract = getContract({ address: vault, abi: redemptionVaultAbi, client: encoder.client });`, then something like:

```suggestion
        midasContract.read.minAmount(),
        midasContract.read.instantFee(),
        midasContract.read.instantDailyFee(),
        midasContract.read.STABLECOIN_RATE(),
        midasContract.read.waivedFeeRestriction(encoder.address),
        midasContract.read.dailyLimits(getTodayAsNumber()),
        midasContract.read.mTokenDataFeed(),
        midasContract.read.tokensConfig(tokenOut),
```